web-api/src/ingestion/routes.rs

```diff
@@ -159,6 +193,11 @@ impl UnvalidatedIngestedDocument {
     fn validate(self) -> Result<IngestedDocument, DocumentIdAsObject> {
         let validate = || -> anyhow::Result<_> {
             let id = self.id.as_str().try_into()?;
+            let snippet = if self.summarize {
+                Snippet::Raw(self.snippet)
```

Comment:
The summarizer is returning `Result<String>` atm but it cannot really fail. I can change that and I would be more to call it here than having `Snippet`. In the end this is already a transformation and not only validation.
wdyt?

Reply:
i wasn't sure if we should put the summarization into the validation, but since you're ok with that i'll change it as it simplifies the code.

Reply:
Would be better to have validation and transformation as separate steps but it will add overhead. Since the summarizer cannot trigger an error I think is better to have it here than to be forced to always have mut document to be able to read the snippet.